pyaedt/edb_grpc/core/components.py

```diff
@@ -0,0 +1,2209 @@
+            Whether the RLC component is parallel. The default is ``False``.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        Examples
+        --------
+
+        >>> from pyaedt import Edb
+        >>> edbapp = Edb("myaedbfolder")
+        >>> edbapp.core_components.set_component_rlc(
+        ...     "R1", res_value=50, ind_value=1e-9, cap_value=1e-12, isparallel=False
+        ... )
+
+        """
+        edb_component = self.get_component_by_name(componentname)
+        edb_rlc_component_property = self._edb.Cell.Hierarchy.RLCComponentProperty()
+        component_pins = self.get_pin_from_component(componentname)
+        pin_number = len(component_pins)
+        if pin_number == 2:
+            from_pin = component_pins[0]
+            to_pin = component_pins[1]
+            rlc = self._edb.Utility.Rlc()
+            rlc.IsParallel = isparallel
+            if res_value is not None:
+                rlc.REnabled = True
+                rlc.R = self._get_edb_value(res_value)
+            else:
+                rlc.REnabled = False
+            if ind_value is not None:
+                rlc.LEnabled = True
+                rlc.L = self._get_edb_value(ind_value)
+            else:
+                rlc.LEnabled = False
+            if cap_value is not None:
+                rlc.CEnabled = True
+                rlc.C = self._get_edb_value(cap_value)
+            else:
+                rlc.CEnabled = False
+            pin_pair = self._edb.Utility.PinPair(from_pin.GetName(), to_pin.GetName())
+            rlc_model = self._edb.Cell.Hierarchy.PinPairModel()
+            rlc_model.SetPinPairRlc(pin_pair, rlc)
+            if not edb_rlc_component_property.SetModel(rlc_model) or not edb_component.SetComponentProperty(
+                edb_rlc_component_property
+            ):
+                self._logger.error("Failed to set RLC model on component")
+                return False
+        else:
+            self._logger.warning(
+                "Component %s has not been assigned because either it is not present in the layout "
+                "or it contains a number of pins not equal to 2",
+                componentname,
+            )
+            return False
+        self._logger.warning("RLC properties for Component %s has been assigned.", componentname)
+        return True
+
+    @pyaedt_function_handler()
+    def update_rlc_from_bom(
+        self,
+        bom_file,
+        delimiter=";",
+        valuefield="Func des",
+        comptype="Prod name",
+        refdes="Pos / Place",
+    ):
+        """Update the EDC core component values (RLCs) with values coming from a BOM file.
+
+        Parameters
+        ----------
+        bom_file : str
+            Full path to the BOM file, which is a delimited text file.
+            Header values needed inside the BOM reader must
+            be explicitly set if different from the defaults.
```

Comment:
```suggestion
            Header values needed inside the BOM reader must
            be explicitly set if they are different from the defaults.
```
We want to use full sentences after the first line with the brief description.